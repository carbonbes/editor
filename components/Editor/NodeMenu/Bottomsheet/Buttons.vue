<template>
  <div class="px-6 pb-6 w-full flex flex-col gap-2">
    <slot />
  </div>
</template>
