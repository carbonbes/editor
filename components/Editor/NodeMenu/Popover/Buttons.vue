<template>
  <div class="flex gap-1">
    <slot />
  </div>
</template>
