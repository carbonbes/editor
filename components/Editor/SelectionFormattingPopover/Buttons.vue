<template>
  <BaseFloating class="p-1 rounded-2xl">
    <div class="flex items-center gap-1">
      <slot />
    </div>
  </BaseFloating>
</template>

<script setup lang="ts">
import BaseFloating from '~/components/Shared/BaseFloating.vue'
</script>
