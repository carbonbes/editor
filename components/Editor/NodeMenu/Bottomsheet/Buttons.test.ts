import { describe, expect, test } from 'vitest'
import Buttons from './Buttons.vue'
import { mount } from '@vue/test-utils'

describe('Buttons', () => {
  test('should be defined', () => {
    expect(Buttons).toBeDefined()
  })

  describe('correctly displayed', () => {
    function mountButtons(slot?: string) {
      const Component = defineComponent({
        components: { Buttons },
        template: `
          <Buttons>${slot || ''}</Buttons>
        `,
      })

      return mount(Component)
    }

    const cases = [
      ['without a slot'],
      ['with a slot', '<p>Test</p>'],
    ]

    test.each(cases)('%s', async (_, slot) => {
      const wrapper = mountButtons(slot)

      expect(wrapper.html()).toMatchSnapshot()

      wrapper.unmount()
    })
  })
})
