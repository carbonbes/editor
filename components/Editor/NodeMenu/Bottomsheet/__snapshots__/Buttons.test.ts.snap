// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Buttons > correctly displayed > with a slot 1`] = `
"<div class="px-6 pb-6 w-full flex flex-col gap-2">
  <p>Test</p>
</div>"
`;

exports[`Buttons > correctly displayed > without a slot 1`] = `"<div class="px-6 pb-6 w-full flex flex-col gap-2"></div>"`;
